### PR TITLE
Fix null pointer deref in continuous aggregates

### PR DESCRIPTION
**Pull request by mkindahl**
There is a potential null pointer dereference in that `raw_hypertable`
might be NULL when counting the number of continuous aggregates
attached. This commit fixes this by assuming that no continuous
aggregates are attached if `raw_hypertable` is NULL.

Found using `clang-tidy`.

### CODE REVIEW

File: src/continuous_agg.c [hunk 1]
```diff
@@ -523,7 +523,8 @@ drop_continuous_agg(ContinuousAgg *agg, bool drop_user_view)
 	/* lock catalogs */
 	LockRelationOid(catalog_get_table_id(catalog, BGW_JOB), RowExclusiveLock);
 	LockRelationOid(catalog_get_table_id(catalog, CONTINUOUS_AGG), RowExclusiveLock);
-	raw_hypertable_has_other_caggs = number_of_continuous_aggs_attached(raw_hypertable->fd.id) > 1;
+	raw_hypertable_has_other_caggs =
```
Comment (erimatnor) [inline src/continuous_agg.c:526]: While this seems to solve the problem with minimal changes, I think this code is hard to follow (and potentially error prone) because it rechecks the same conditions (`raw_hypertable_exists` and `raw_hypertable_has_other_caggs`) in multiple places. It would make more sense to group the conditions together unless the ordering of operations are important here.

So, while a refactor might be out-of-scope here, I would consider creating an issue to refactor this and tag it `refactor`.

Comment (mkindahl) [reply src/continuous_agg.c:526]: I agree with that. I found the logic hard to follow and hence opted for a simpler fix. Added an issue in the private repository.
